File: typescript/base-hmi-types/src/screens/screen/HmiScreenWindow.ts
import { HmiProperty } from "../base/HmiProperty.js";
import { HmiWindowBase } from "../base/HmiWindowBase.js";
import { HmiObjectType } from "../base/HmiObjectType.js";

export class HmiScreenWindow extends HmiWindowBase {
  constructor() {
    super();
    this.hmiObjectType = HmiObjectType.HmiScreenWindow;
  }

  screenId?: HmiProperty<string>;
  screenName?: HmiProperty<string>;
  tabIntoWindow?: HmiProperty<boolean>;
  startupPosition?: HmiProperty<number>;
  windowState?: HmiProperty<number>;
  isModal?: HmiProperty<boolean>;
  offsetLeft?: HmiProperty<number>;
  offsetTop?: HmiProperty<number>;
}

File: src/BaseHmiTypes/Screens/HmiScreenWindow.cs
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Screens;

public class HmiScreenWindow : HmiWindowBase
{
    public HmiScreenWindow()
    {
        HmiObjectType = BaseHmiTypes.Screens.Base.HmiObjectType.HmiScreenWindow;
    }

    public HmiProperty<string>? ScreenId { get; set; }

    public HmiProperty<string>? ScreenName { get; set; }

    public HmiProperty<bool>? TabIntoWindow { get; set; }

    public HmiProperty<int>? StartupPosition { get; set; }

    public HmiProperty<int>? WindowState { get; set; }

    public HmiProperty<bool>? IsModal { get; set; }

    public HmiProperty<double>? OffsetLeft { get; set; }

    public HmiProperty<double>? OffsetTop { get; set; }
}
